### ISSUE TITLE
Common methods per widget

### ISSUE THREAD

**Issue by mjlbach** [mjlbach/neovim-ui#5]
It came up on IRC that we should have a common interface for widgets. This is mostly a placeholder for discussion.

-[ ] widget.close() -> Should we also enable a delayed invocation of this as a widget method like I did in notification.lua?
-[ ] widget.id() -> @smolck do we need this for vim.ui.layout?

**Comment by smolck** [comment]
>  -[ ] widget.id() -> @smolck do we need this for vim.ui.layout?

I don't think so, but so I understand, what would be the purpose of this in general?

**Comment by mjlbach** [comment]
The purpose of the metatable/common method guidelines, or the purpose of having an "ids" method?

**Comment by smolck** [comment]
> The purpose of the metatable/common method guidelines, or the purpose of having an "ids" method?

I meant the purpose of the `id()` method (which I guess `ui.layout` could build off of if it ends up being something we want/need in `ui.layout`).
